crates/bevy_wgpu/src/wgpu_renderer.rs

```diff
@@ -47,7 +47,12 @@ impl WgpuRenderer {
             .expect("Unable to find a GPU! Make sure you have installed required drivers!");
 
         #[cfg(feature = "trace")]
-        let trace_path = Some(std::path::Path::new("wgpu_trace"));
+        let trace_path = {
+            let path = std::path::Path::new("wgpu_trace");
+            // ignore potential error, wgpu will log it
+            let _ = std::fs::create_dir(path);
+            Some(path)
+        };
         #[cfg(not(feature = "trace"))]
         let trace_path = None;
```

Comment:
This whole block could be cobined into a `if cfg!(...)` expression which would be clearer imo.
```rust
let trace_path = if cfg!(feature = "trace") {
    let path = std::path::Path::new("wgpu_trace");
    // ignore potential error, wgpu will log it
    let _ = std::fs::create_dir(path);
    Some(path)
} else {
    None
};
```

Reply:
I prefer the current way as I feel it's less work for the compiler.

But then this should be very simple to compile away... 🤷